core/variant_op.cpp

```diff
@@ -161,35 +162,78 @@ bool Variant::booleanize(bool &r_valid) const {
 			return _data._real;
 		case STRING:
 			return (*reinterpret_cast<const String *>(_data._mem)) != "";
-		case VECTOR2:
-		case RECT2:
-		case TRANSFORM2D:
-		case VECTOR3:
-		case PLANE:
-		case RECT3:
-		case QUAT:
-		case BASIS:
-		case TRANSFORM:
-		case COLOR:
+		case VECTOR2: {
+			const Vector2 *v = reinterpret_cast<const Vector2 *>(_data._mem);
+			return v->x + v->y;
+		}
+		case RECT2: {
+			const Rect2 *v = reinterpret_cast<const Rect2 *>(_data._mem);
+			return v->position.x + v->position.y + v->size.x + v->size.y;
```

Comment:
That's also wrong logic, but what should be done instead is up for discussion. What would be a "false" Rect2?
I guess we have two possibility:
- `position` is false (Point2(0,0)) and `size` is false (Vector2(0,0))
- `size` is false, regardless of the position

I guess the former is likely more consistent (like for other types, there would be only one false case), even if one could argue that an empty Rect2 at position (100, 213) could be "false".